From bf17c79171512d6a27c31e003e60a25e7a093226 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alberts=20Muktup=C4=81vels?= <alberts.muktupavels@gmail.com>
Date: Thu, 12 Mar 2020 15:16:59 +0200
Subject: [PATCH] window: do not unfocus on new window

It doesn't make much sense to globally unset focus when mapping
a new window.

Based on mutter commit:
https://gitlab.gnome.org/GNOME/mutter/-/commit/998d921d418a7e943ea
---
 src/core/window.c | 18 +-----------------
 1 file changed, 1 insertion(+), 17 deletions(-)

diff --git a/src/core/window.c b/src/core/window.c
index 31226c4e..ff2c34a1 100644
--- a/src/core/window.c
+++ b/src/core/window.c
@@ -2200,23 +2200,7 @@ meta_window_show (MetaWindow *window)
       ( (!place_on_top_on_map && !takes_focus_on_map) ||
       will_be_covered )
     ) {
-      if (meta_window_is_ancestor_of_transient (focus_window, window))
-        {
-          /* This happens for error dialogs or alerts; these need to remain on
-           * top, but it would be confusing to have its ancestor remain
-           * focused.
-           */
-          meta_topic (META_DEBUG_STARTUP,
-                      "The focus window %s is an ancestor of the newly mapped "
-                      "window %s which isn't being focused.  Unfocusing the "
-                      "ancestor.\n",
-                      focus_window->desc, window->desc);
-
-          meta_display_focus_the_no_focus_window (window->display,
-                                                  window->screen,
-                                                  timestamp);
-        }
-      else
+      if (!meta_window_is_ancestor_of_transient (focus_window, window))
         {
           needs_stacking_adjustment = TRUE;
           if (!window->placed)
-- 
2.24.1

